lib/active_merchant/billing/gateways/payeezy.rb

```diff
@@ -307,8 +307,8 @@ def add_stored_credentials(params, options)
       end
 
       def original_transaction_id(options)
-        return options[:cardbrand_original_transaction_id] if options[:cardbrand_original_transaction_id]
-        return options[:stored_credential][:network_transaction_id] if options.dig(:stored_credential, :network_transaction_id)
+        network_transaction_id = options[:cardbrand_original_transaction_id] || options.dig(:stored_credential, :network_transaction_id)
+        return options[:cardbrand_original_transaction_id] = network_transaction_id if network_transaction_id
```

Comment:
I'm a little confused if/why setting `options[:cardbrand_original_transaction_id]` here is necessary, instead of just returning `network_transaction_id` at this point?

Reply:
You're right, let me fix that

Reply:
@curiousepic done!